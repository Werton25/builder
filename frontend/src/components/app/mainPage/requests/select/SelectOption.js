import React from 'react';
import PropTypes from 'prop-types';
import {deleteRequest} from "../../../../../actions/requests";
import config from "../../../../../config";
import {connect} from 'react-redux';
import {getCurrentUri} from "../../../../../selectors/router";
import {getCurrentRequest} from "../../../../../selectors/requests";
import {changeSelected} from "../../../../../actions/select";

export class BaseSelectOption extends React.Component {
  handleMouseDown = (event) => {
    event.preventDefault();
    event.stopPropagation();
    this.props.onSelect(this.props.option, event);
  };

  handleMouseEnter = (event) => {
    this.props.onFocus(this.props.option, event);
  };

  handleMouseMove = (event) => {
    this.props.onFocus(this.props.option, event);
  };

  render() {
    const {
      option,
      deleteRequest,
      currentUri,
      currentRequest,
      changeSelected
    } = this.props;
    return (
      <div className={this.props.className} title={option.title}>
        <span onMouseDown={this.handleMouseDown} onMouseMove={this.handleMouseMove}
             onMouseEnter={this.handleMouseEnter}>
          {option.name}
        </span>
        {
          option.id !== config.defaultRequestId &&
          <i role='button' className='glyphicon glyphicon-remove text-danger pull-right'
             onClick={() => {
               currentRequest.id === option.id && changeSelected(currentUri, config.defaultRequestId);
               deleteRequest(currentUri, option.id);
             }}/>
        }
      </div>
    );
  }
}

BaseSelectOption.propTypes = {
  currentUri: PropTypes.string.isRequired,
  currentRequest: PropTypes.object.isRequired,
  changeSelected: PropTypes.func.isRequired,
  deleteRequest: PropTypes.func.isRequired
};

const mapStateToProps = (state) => ({
  currentUri: getCurrentUri(state),
  currentRequest: getCurrentRequest(state)
});

const mapDispatchToProps = {
  changeSelected,
  deleteRequest
};

export default connect(mapStateToProps, mapDispatchToProps)(BaseSelectOption);
