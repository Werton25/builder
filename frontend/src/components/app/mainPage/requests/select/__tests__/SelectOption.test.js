import React from "react";
import renderer from 'react-test-renderer';
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import SelectOption from "../SelectOption";

const store = configureStore()({
  router: {location: {pathname: '/service1/method1'}},
  requests: {
    '/service1/method1': {
      'default': {
        id: 'default',
        name: 'default',
        value: 'test',
        schema: 'test'
      }
    }
  },
  select: {}
});

describe('SelectOption tests', () => {
  it('SelectOption renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <SelectOption option={{
            id: 'default',
            name: 'default',
            value: 'test',
            schema: 'test'
          }}/>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
