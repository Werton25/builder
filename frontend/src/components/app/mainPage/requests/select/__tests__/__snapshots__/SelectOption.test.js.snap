// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectOption tests SelectOption renders correctly 1`] = `
<div
  className={undefined}
  title={undefined}
>
  <span
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseMove={[Function]}
  >
    default
  </span>
</div>
`;
